pkgs/development/compilers/squeak/default.nix

```diff
@@ -1,33 +1,195 @@
-{ lib, stdenv, fetchurl, cmake, coreutils, dbus, freetype, glib, gnused
-, libpthreadstubs, pango, pkg-config, libpulseaudio, which }:
+{ lib, stdenv, fetchFromGitHub, fetchurl, fetchzip
+, autoconf, automake, autoreconfHook, bash, clang, dos2unix, file, gzip, perl
+, pkg-config
+, alsaLib, coreutils, freetype, glib, glibc, gnugrep, libpulseaudio, libtool
+, libuuid, openssl, pango, xorg
+, squeakImageVersion ? null, squeakImageHash ? null
+, squeakSourcesHash ? null, squeakSourcesVersion ? null, squeakVersion ? null
+, squeakVmVersion ? null, squeakVmVersionRelease ? null
+, squeakVmCommitHash ? null, squeakVmCommitHashHash ? null
```

Comment:
Do we really need this mess?

Reply:
This is the sanest way I could think of to not make version overrides require copying & pasting this source into a new expression.

Reply:
> This is the sanest way I could think of to not make version overrides require copying & pasting this source into a new expression.

you could put the "builder" in a separate nix file

Reply:
> you could put the "builder" in a separate nix file

How does this allow overrides, though? If I have a `./generic.nix` with an attrset parameter taking these values, returning a `{ lib, stdenv, ... }: stdenv.mkDerivation { ... }` expression, that's most likely not going to be exported at top-level. It would probably be paired with a `./default.nix` that is `import ./generic.nix { squeakVersion = "5.3-19459"; ... }`, and this doesn't address the concerns of overriding the version without having to paste the whole expression source. nixpkgs/lib's module system addresses this sort of customization. `callPackage` doesn't.

I considered doing something with `makeScope`, but that seemed like more of an annoyance than using a few more named arguments.